zerver/openapi/zulip.yaml

```diff
@@ -383,10 +383,13 @@ paths:
                               `{server_url}/user_uploads/{realm_id}/ab/cdef/temp_file.py`.
                           size:
                             type: integer
+                            nullable: true
                             description: |
                               Size of the file in bytes.
                           create_time:
-                            type: integer
+                            oneOf:
+                            - type: number
+                            - type: integer
```

Comment:
This is a datetime type; should it just be number?

Reply:
It was integer in the example hence I added oneOf. Should I change the example and remove oneOf?

Reply:
I think this might be a situation where we should change the actual code to wrap `int()` where we marshall these in `models.py` so that this is always an integer; can you try submitting a little PR to do that as a followup?

Reply:
I think that only the example is wrong in this case. Adding a `.0` to the example and then removing the oneOf passed the tests. I can convert it to the `int` but then that would be a significant decision as we use `floatify_date_time_field` for all datetime fields. So what should be done?

Reply:
I don't think `floatify_date_time_field` is relevant; to be clear I'm only talking about changing this:
```
class Attachment(AbstractAttachment):                                                                           
    messages: Manager = models.ManyToManyField(Message)                                                         
                                                                                                                
    def is_claimed(self) -> bool:                                                                               
        return self.messages.count() > 0                                                                        
                                                                                                                
    def to_dict(self) -> Dict[str, Any]:                                                                        
        return {                                                                                                
            'id': self.id,                                                                                      
            'name': self.file_name,                                                                             
            'path_id': self.path_id,                                                                            
            'size': self.size,                                                                                  
            # convert to JavaScript-style UNIX timestamp so we can take                                         
            # advantage of client timezones.                                                                    
            'create_time': time.mktime(self.create_time.timetuple()) * 1000,                                    
            'messages': [{                                                                                      
                'id': m.id,                                                                                     
                'name': time.mktime(m.date_sent.timetuple()) * 1000,                                            
            } for m in self.messages.all()],                                                                    
        }                                                                                                       
                                                                                                                
```

to wrap the `create_time` block with `int()` so that we're not using a float to accidentally show microseconds.  While we're at it, we should also fix the super weird name for the timestamps for the messages (or maybe delete that entirely; not sure if it's used)?